extensions/binding/org.eclipse.smarthome.binding.fsinternetradio/src/main/java/org/eclipse/smarthome/binding/fsinternetradio/internal/FSInternetRadioHandlerFactory.java

```diff
@@ -35,13 +35,6 @@ public boolean supportsThingType(ThingTypeUID thingTypeUID) {
 
     @Override
     protected ThingHandler createHandler(Thing thing) {
-
-        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-
-        if (thingTypeUID.equals(THING_TYPE_RADIO)) {
-            return new FSInternetRadioHandler(thing);
-        }
-
-        return null;
+        return new FSInternetRadioHandler(thing);
```

Comment:
I agree that the framework should call this function only if the thing type is part of the supported ones.
@kaikreuzer If there is only one supported thing type, I think we could rely on the fact that the framework calls this function only if the thing consists of that respective type.
We also trust that the OSGi framwork calls our functions (activate, bind, etc.) with the correct arguments.